code/datums/gamemodes/crash.dm

```diff
@@ -0,0 +1,243 @@
+	GLOB.latejoin_gateway = shuttle.spawnpoints
+
+	// Create xenos
+	for(var/i in xenomorphs)
+		var/datum/mind/M = i
+		if(M.assigned_role == ROLE_XENO_QUEEN)
+			transform_queen(M)
+		else
+			transform_xeno(M)
+
+
+#define CRASH_DRAW (1 << 0)
+#define CRASH_XENO_MAJOR (1 << 1)
+#define CRASH_XENO_MINOR (1 << 2)
+#define CRASH_MARINE_MINOR (1 << 3)
+#define CRASH_MARINE_MAJOR (1 << 4)
+
+
+/datum/game_mode/crash/check_finished()
+	if(round_finished)
+		return TRUE
+
+	if(world.time < (SSticker.round_start_time + 15 SECONDS))
+		return FALSE
+
+	var/list/living_player_list = count_humans_and_xenos()
+	var/num_humans = living_player_list[1]
+	var/num_xenos = living_player_list[2]
+	
+	var/list/grounded_living_player_list = count_humans_and_xenos(SSmapping.levels_by_any_trait(list(ZTRAIT_GROUND)))
+	var/num_grounded_humans = grounded_living_player_list[1]
+	
+
+	var/victory_options = (num_humans == 0 && num_xenos == 0 || (planet_nuked && !marines_evac)) 		<< 0 // Draw, for all other reasons
+	victory_options |= (!planet_nuked && num_humans == 0 && num_xenos > 0) 								<< 1 // XENO Major (All marines killed)
+	victory_options |= (marines_evac && !planet_nuked) 													<< 2 // XENO Minor (Marines evac'd )
+	victory_options |= (marines_evac && planet_nuked && (num_humans == 0 || num_grounded_humans > 0)) 	<< 3 // Marine minor (Planet nuked, some human left on planet)
+	victory_options |= (marines_evac && planet_nuked && num_grounded_humans == 0) 						<< 4 // Marine Major (Planet nuked, marines evac)
+	switch(victory_options)
+		if(CRASH_DRAW)
+			message_admins("Round finished: [MODE_CRASH_DRAW_DEATH]")
+			round_finished = MODE_CRASH_DRAW_DEATH 
+		if(CRASH_XENO_MAJOR)
+			message_admins("Round finished: [MODE_CRASH_X_MAJOR]")
+			round_finished = MODE_CRASH_X_MAJOR 
+		if(CRASH_XENO_MINOR)
+			message_admins("Round finished: [MODE_CRASH_X_MINOR]")
+			round_finished = MODE_CRASH_X_MINOR 
+		if(CRASH_MARINE_MINOR)
+			message_admins("Round finished: [MODE_CRASH_M_MINOR]")
+			round_finished = MODE_CRASH_M_MINOR 
+		if(CRASH_MARINE_MAJOR)
+			message_admins("Round finished: [MODE_CRASH_M_MAJOR]")
+			round_finished = MODE_CRASH_M_MAJOR 
+
+	return FALSE
+
+/datum/game_mode/crash/declare_completion()
+	. = ..()
+	to_chat(world, "<span class='round_header'>|[round_finished]|</span>")
+	to_chat(world, "<span class='round_body'>Thus ends the story of the brave men and women of the [CONFIG_GET(string/ship_name)] and their struggle on [SSmapping.config.map_name].</span>")
+	
+	SEND_SOUND(world, GAMEMODE_CRASH_MUSIC[round_finished])
+	
+	log_game("[round_finished]\nGame mode: [name]\nRound time: [duration2text()]\nEnd round player population: [length(GLOB.clients)]\nTotal xenos spawned: [GLOB.round_statistics.total_xenos_created]\nTotal humans spawned: [GLOB.round_statistics.total_humans_created]")
+
+	announce_medal_awards()
+	announce_round_stats()
+	end_of_round_deathmatch()
+
+
+/datum/game_mode/crash/proc/initialize_xeno_leader()
```

Comment:
To avoid duplicate procs `/crash` and `/distress` could be a child of the same intermediate parent, something like `/xenomorph` or the likes.

Reply:
Ideally the proc should just get moved to gamemode if they become shared